### DISCUSSION TITLE
Strange results with log_logit link settings

### DISCUSSION THREAD

**Discussion by abelowska** [lnccbrown/HSSM#374] [Q&A] [answered]
Hi,

I'm fitting the hierarchical full DDM model on Flanker data. I'm using safe priors with the `log_logit` link setting and (default) `blackbox` likelihood function.  I've pasted model specification below:

```
hssm.HSSM(
    data=data_df,
    model='full_ddm',
    include=[
        {
            "name": "a",
            "formula": "a ~ 1 + (1|participant_id)",
        },
        {
            "name": "v",
            "formula": "v ~ 1 + (1|participant_id)",
        },
        {
            "name": "t",
            "formula": "t ~ 1 + (1|participant_id)",
        },
        
    ],
    z=0.5,
    sz=0,
    sv=0,
    st=0,
    prior_settings="safe",
    link_settings = "log_logit"
)
```

Unfortunately, I've got strange results:
- non-decision time does not converge
- `a_Intercept` value is very close to 0 (looks like issue similar to: https://github.com/lnccbrown/HSSM/discussions/346)
- random effects for `a` are negative

![Screenshot from 2024-03-13 18-31-33](https://github.com/lnccbrown/HSSM/assets/26854113/569d48ce-f273-42e8-abb4-ece048223873)

I'm quite surprised because it is not very complicated model and I successfully fitted it with Stan.

When I bounded tau to `(0.05, 0.4)` , trying to force chains to not explore super small tau-s, the results are quite similar: 

- `a_Intercept` value is very close to 0
- random effects for `a` are negative
- random effects for `t` are negative

![Screenshot from 2024-03-13 18-33-33](https://github.com/lnccbrown/HSSM/assets/26854113/ea54b3fa-8cc0-437e-b314-8a4449731668)

These results make me think that the estimates of `a` and `t` are not in probability space but rather log/logit space. Can you confirm that? And if so, can you recommend how to properly recover estimates of `t` and individual's `t` (and theoretical other regression parameters)?

Appreciate any help!

**Comment by AlexanderFengler** [comment, marked as answer]
Hi @abelowska,

Two things, 

First, in fact, we figured that there was a bug, effectively messing up prior settings when using the `log_logit` links. We are working to fix that, aiming for release by the end of the weekend.

The chain starting points are very suboptimal for some of the parameters when using `log_logit` setting and some unnecessary parameter limits are maintained. 

Second, indeed, when applying the `log_logit` setting, the parameters are in logit space, so in principle, for better interpretation you may want to push them through the corresponding sigmoid. 

I will add an issue to include more convenience functionality concerning this case, since it seems that we are, for now, focusing more on the `log_logit` case than we a priori anticipated.

Best,
Alex

**Reply by abelowska** [reply]
Thank for the reply. I'm looking forward to the release!

**Comment by eort** [comment]
Hi @AlexanderFengler , 

Just to confirm, that fix has not been released yet? I observe the same issue as @abelowska.

>  for better interpretation you may want to push them through the corresponding sigmoid.

What exactly do you mean with "corresponding"? 

Thanks!
